_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,419 @@
+```
+Increase the number of nodes with `doctl`:
+
+```bash
+doctl k cluster node-pool update do1 do1-default-pool --count 4
+```
+
+### Increase Storage
+
+For the `store-mongodb` stateful set with 3 replicas, some of the pods did not run due to _unbound immediate PersistentVolumeClaims_. When inspecting the pod events, I found out there was a failure in volume provisioning:
+
+```text
+Events:
+  Type     Reason                Age                  From                                                                       Message
+  ----     ------                ----                 ----                                                                       -------
+  Normal   Provisioning          3m6s (x10 over 11m)  dobs.csi.digitalocean.com_master-do3_cebb3451-5eba-4805-956c-753ec148e2ea  External provisioner is provisioning volume for claim "demo/datadir-store-mongodb-2"
+  Warning  ProvisioningFailed    3m5s (x10 over 11m)  dobs.csi.digitalocean.com_master-do3_cebb3451-5eba-4805-956c-753ec148e2ea  failed to provision volume with StorageClass "do-block-storage": rpc error: code = ResourceExhausted desc = volume limit (10) has been reached. Current number of volumes: 10. Please contact support.
+  Normal   ExternalProvisioning  103s (x43 over 11m)  persistentvolume-controller                                                waiting for a volume to be created, either by external provisioner "dobs.csi.digitalocean.com" or manually created by system administrator
+```
+
+As instructed by the event message, I contacted DigitalOcean support and they fixed it.
+
+### Find your gateway's external IP and update redirect URIs
+
+Once all the pods are running, find the gateway external IP with the `kubectl describe` command:
+
+```bash
+kubectl describe service gateway -n demo
+```
+
+```text
+Name:                     gateway
+Namespace:                demo
+Labels:                   app=gateway
+Annotations:              kubernetes.digitalocean.com/load-balancer-id: e81d2b8c-6c28-430d-8ba8-6dab29a1ba76
+                          service.beta.kubernetes.io/do-loadbalancer-certificate-id: bd0b1d03-0f90-449d-abbe-ac6a4026c133
+Selector:                 app=gateway
+Type:                     LoadBalancer
+IP Family Policy:         SingleStack
+IP Families:              IPv4
+IP:                       10.245.47.51
+IPs:                      10.245.47.51
+LoadBalancer Ingress:     157.230.200.181
+Port:                     http  8080/TCP
+TargetPort:               8080/TCP
+NodePort:                 http  31048/TCP
+Endpoints:                10.244.1.42:8080
+Session Affinity:         None
+External Traffic Policy:  Cluster
+```
+
+Update the redirect URIs in Okta to allow the gateway address as a valid redirect. Run `okta login`, open the returned URL in your browser, and sign in to the Okta Admin Console. Go to the **Applications** section, find your application, and edit it.
```

Comment:
I think you might need to be more explicit here. I added `http://157.230.200.181` and didn't realize I needed to add `http://157.230.200.181:8080` so the first time I logged in, I got an error and had to adjust my redirect URIs after inspecting the value in my browser.

<img width="1129" alt="Screen Shot 2022-05-18 at 11 37 25" src="https://user-images.githubusercontent.com/17892/169030033-aec057ea-2c9d-4526-aa64-62b52a17959d.png">

Reply:
Added the explicit URIs to the post.